lib/plugins/aws/package/compile/functions/index.js

```diff
@@ -537,52 +537,88 @@ class AwsCompileFunctions {
         versionCompilationPromise = BbPromise.resolve();
       }
 
-      return versionCompilationPromise.then(() => this.compileFunctionDestinations(functionName));
+      return versionCompilationPromise.then(() =>
+        this.compileFunctionEventInvokeConfig(functionName)
+      );
     });
   }
 
-  compileFunctionDestinations(functionName) {
+  compileFunctionEventInvokeConfig(functionName) {
     const functionObject = this.serverless.service.getFunction(functionName);
-    const { destinations } = functionObject;
-    if (!destinations) return;
+    const {
+      destinations,
+      maximumEventAge: maximumEventAgeOnFunction,
+      maximumRetryAttempts: maximumRetryAttemptsOnFunction,
+    } = functionObject;
+    let maximumEventAgeOnDestinations;
+    let maximumRetryAttemptsOnDestinations;
+
+    if (!destinations && !maximumEventAgeOnFunction && !maximumRetryAttemptsOnFunction) return;
+
+    const destinationConfig = {};
+
+    if (destinations) {
+      if (destinations.maximumRetryAttempts !== undefined) {
```

Comment:
Let's write it as: `if (destinations.maximumRetryAttempts != null)` (naturally we should discard both `null` and `undefined`